### DISCUSSION TITLE
Github pages isn't working after switching from public to private and then back to public

### DISCUSSION THREAD

**Discussion by aagontuk** [community/community#23647] [Other Feature Feedback, Questions, & Ideas]
<p>I am using the free plan of github. After hearing the unlimited private repositories for free plans news, I thought I could make my <a href="https://aagontuk.github.io/" rel="nofollow noopener">personal website</a> repo private. So I made the repo private. But then I noticed, github pages to work under private repo, you have to subscribe to github pro plans. So I revert back to public. But now my site isn’t working. How can I re-enable my site?</p>
<p><strong>update:</strong> My site is working because as I didn’t get any solution, I have deleted my old repository and recreated it. Still I am waiting for a good solution.</p>

**Comment by lee-dohm** [comment]
<p>Unfortunately, we can’t make a guess as to what might have gotten wedged that was preventing this from working. If anyone else runs into this, please reach out to GitHub Support at <a href="https://github.com/contact">https://github.com/contact</a> so they can take a deeper look at what the issue might be.</p>
<p><strong>NB: Response edited December 2020 to reflect changes in GitHub’s support channels.</strong></p>

**Comment by andychung** [comment]
<p>This happened to me as well - I’ve also emailed support in hopes of a solution.</p>

**Comment by andychung** [comment]
<p>This problem seems to resolve itself if you push any commit to your pages branch.</p>

**Reply by altnomerus** [reply]
This was helpful to me, and I appreciate that.

**Comment by kxdang** [comment]
<p>What about master branch? I tried to make my repo public and did a push to master branch (only branch I have)</p>

**Comment by Ferido07** [comment]
<p>It did work for me after pushing to master branch</p>

**Comment by abt86croz** [comment]
<p>Any thoughts on how I can remedy this situation? I already sent an email to support.</p>
<p>I turned the repo to private, realized the page went 404. Made the repo public again and the 404 remains.</p>
<p>It looks like the UI settings where GitHub pages (page) use to be has changed considerably. I don’t see any option to republish the page and the option to redirect my custom domain url.</p>

**Comment by abt86croz** [comment]
<p>Plus one. Were you able to fix this issue? My page is down. I did the exact same thing as you. Also, I don’t see an option to republish and add a redirect under settings no more.</p>

**Comment by pankajpj07** [comment]
<p>Thanks Bro. It also worked for me after pushing to the main branch. :slight_smile:</p>

**Comment by saimunhossain** [comment]
<p>I have solved after pushing a commit to master branch</p>

**Comment by jubairmizan** [comment]
<p>Dude You free me from tensed… Thanks Man :slight_smile:</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi friends, </p>
<p>Updating this thread to share a solution that has worked for some, please try mind pushing a commit to your repository to trigger a build, this should take care of the issue. </p>
<p>I hope this helps!-</p>

**Comment by Prashavu** [comment]
<p>Hi</p>
<p>The same exact thing happened to me too today.</p>
<p>I initially thought making a personal/user github page repo would  enable my pages public even though repo can be set to private.Looks like it is not so and is only availble for Github Pro</p>
<p>So now I have set back to Public again and "Your page is published at “URL” is missing rom the section in options under settings</p>
<p>Pls correct this Github</p>

**Comment by rodneycurl** [comment]
<p>This happened to me too. Here’s how I fixed it.</p>
<p>1 Go to the ‘repository page’</p>
<p>2 Click on ‘settings’ </p>
<p>3 Scroll down to ‘Github Pages’</p>
<p>4 Click on the dropdown for ‘Source’</p>
<p>5 Select ‘master branch’</p>
<p>Then refresh your website and it should work.</p>

**Reply by kittycat-tech** [reply]
thanks! I did this , then renamed the repo and my website is working now! :)

**Reply by JonSnowey** [reply]
This did the trick! Thank you.

**Reply by MarcoBrian** [reply]
confirm this works :) thank you! Just need to trigger the build again

**Reply by magezashalom1** [reply]
> This happened to me too. Here’s how I fixed it.
> 
> 1 Go to the ‘repository page’
> 
> 2 Click on ‘settings’ 
> 
> 3 Scroll down to ‘Github Pages’
> 
> 4 Click on the dropdown for ‘Source’
> 
> 5 Select ‘master branch’
> 
> Then refresh your website and it should work.

--> Thanks fam! This worked for me

**Reply by YeshKrish** [reply]
Thanks a lot, it worked for me.

**Reply by katykat5099** [reply]
It is called just "Pages" now for me, but it is the same solution after that. Thanks

**Reply by ghost** [reply]
Here in 2025 and this worked for me.  I my case the branch was called "main", but worked just as you said.  Thank you!!

![Screen Shot 2025-01-05 at 12 54 35](https://github.com/user-attachments/assets/d2fababc-0efd-4af8-89ed-1178d084b66a)

**Comment by pedestriantactics** [comment]
<p>Going to add to this. I tried re-committing to the master and it didn’t work. However, it did after removing all of my site files, committing, and then adding them back into the repo before committing again.</p>

**Comment by raysome** [comment]
<p>I got the same problem. The solution I tried is to rename the repo, and then rename back. Fortunately, it worked like a charm.</p>

**Comment by hardlyadequate** [comment]
<p>I did this and then I did the rename and rename back. Now it’s working.</p>

**Comment by procodey** [comment]
<p>Bro I only have master branch. I created a random file and committed . Nothing happened even then.</p>
<p>Edit : It worked now lol</p>

**Comment by dohahelmy** [comment]
<blockquote>
<p><a class="mention" href="/andychung">@andychung</a> wrote:</p>
</blockquote>
<blockquote>
<p>This problem seems to resolve itself if you push any commit to your pages branch.</p>
</blockquote>
<p>This worked for me. thank you</p>

**Comment by arcman7** [comment]
<p>This hasn’t been the case for me. My last commit is 18 minutes ago on the master branch. </p>
<p><a href="https://github.com/arcman7/arcman7.github.io" rel="nofollow noopener">https://github.com/arcman7/arcman7.github.io</a></p>
<p>EDIT: While writing this reply I tried committing directly on the remote repository via editing my readme.md file. This seems to have worked as opposed to pushing commits to the remote repository. </p>

**Comment by GorginZ** [comment]
Nothing mentioned here worked for me, I tried renaming repo, committing directly to main, tried committing directly from UI, merging a new PR, but I found an instant fix based on the deploy failure I saw (my builds would all succeed, but the Deployments were not being triggered, the last one being some time before I changed my repo to private then back to public)

Under Deployments, I tried to re-run the deployment steps again for a sanity-check, I re-run my last successful deployment and the failure error was something like can't deploy from gh-pages due to restrictions, so I went to the branch restrictions section in Deployments and added gh-pages as i noticed this was the branch always being deployed from and vaguely remember this "special branch" when I set up pages deployment. Added the deployment branch or tag rule to allow "gh-pages" and retried the build, instant fix.

So When a repo is changed to private I'm imagining one of the things that happens is this dependent rule is removed.
